### PR TITLE
feat: wrap `View`s into a background container

### LINKED ISSUE #3743: Add properties ```gradient```, ```image_src``` for ```View```
### Duplicate Check

- [X] I have searched the [opened issues](https://github.com/flet-dev/flet/issues) and there are no duplicates

### Describe the requested feature

It would be great to have the ability to set an image or gradient as the background directly in the properties of a View, just like you can with a Container.

### PR DESCRIPTION
**Pull request by ndonkoHenri**
## Description

Closes #3743 

## Test Code

```python
import flet as ft


def main(page: ft.Page):
    page.horizontal_alignment = ft.CrossAxisAlignment.CENTER
    page.vertical_alignment = ft.MainAxisAlignment.CENTER
    
    page.bgcolor = ft.colors.TRANSPARENT
    page.decoration = ft.BoxDecoration(
        gradient=ft.LinearGradient(
            colors=[ft.colors.RED, ft.colors.BLUE],
            begin=ft.alignment.top_left,
            end=ft.alignment.bottom_right,
            stops=[0.0, 1.0],
        ),
        image=ft.DecorationImage(
            src="https://images.unsplash.com/photo-1547721064-da6cfb341d50",
            fit=ft.ImageFit.COVER,
            opacity=0.2,
        ),
    )

    page.add(ft.Text("Hello, World!", size=55),)


ft.app(main)

```

## Screenshots (if applicable):
image

## Summary by Sourcery

Add support for background and foreground decorations in View and Container components, enabling more flexible UI designs. Enhance border radius parsing with default values and update macOS client code to use modern Swift attributes.

New Features:
- Introduce background and foreground decoration properties to the View and Container components, allowing for more customizable UI elements.

Enhancements:
- Enhance the borderRadiusFromJSON function to accept a default value, improving flexibility in border radius parsing.
- Update the AppDelegate in the macOS client to use the @main attribute, modernizing the Swift code.

### CODE REVIEW

**Comment by ndonkoHenri** [comment]
> The change in client/macos/Runner/AppDelegate.swift from @NSApplicationMain to @main seems unrelated to the main purpose of this PR. Could you explain the reasoning behind this change?

Yes sir! 🙂
Running the above test code in my dev-environment, Flutter notified me (see below) that `NSApplicationMain` in `AppDelegate.swift` was deprecated, and it automatically made the appropriate modification.
> macos/Runner/AppDelegate.swift uses the deprecated @NSApplicationMain attribute, updating.

File: packages/flet/lib/src/controls/page.dart [hunk 1]
```diff
@@ -1131,14 +1132,28 @@ class _ViewControlState extends State<ViewControl> with FletStoreMixin {
...
+              if (backgroundDecoration != null ||
+                  foregroundDecoration != null) {
+                return Container(
+                  decoration: backgroundDecoration,
+                  foregroundDecoration: foregroundDecoration,
+                  child: result,
+                );
+              }
+              return result;
+            });
```
Comment (FeodorFitsner) [inline packages/flet/lib/src/controls/page.dart:1156]: I'd make wrapping into Container conditional, i.e. when the decoration is specified on Python side. Just to be on the safe side.

Comment (ndonkoHenri) [reply packages/flet/lib/src/controls/page.dart:1156]: Legit 👍🏾

Comment (FeodorFitsner) [reply packages/flet/lib/src/controls/page.dart:1156]: Ah, I see, nvm. I was probably expecting `else` for clarity.

**Review by FeodorFitsner — APPROVED** [review]

**Pull request merged**